torch/export/export_schema.py

```diff
@@ -0,0 +1,319 @@
+#       These are faithful capture of tensor's detail in pytorch's executions during tracing
+#       However, it's up to downstream system on how to utilized these fields
+#       In another word, these feilds are suggestive, rather than mandatory.
+
+@dataclass
+class TensorMeta:
+    dtype: ScalarType
+    sizes: List[SymInt]
+
+    # needed for training
+    requires_grad: bool
+
+    # !!! see description above, there are subtle difference on how these fields should be interpreted
+    device: Device
+    strides: List[SymInt]
+    storage_offset: int
+    layout: Layout
+
+    # !!! memory_format is not tensor property, but it can only used as operator argument
+    # memory_format: MemoryFormat
+
+
+@dataclass
+class Buffer:
+    # !!! TODO: need to define endianness for the buffer
+    buffer: bytes
+
+
+@dataclass
+class ExternalBuffer:
+    # !!! Example from onnx
+    #   // Data can be stored inside the protobuf file using type-specific fields or raw_data.
+    #   // Alternatively, raw bytes data can be stored in an external file, using the external_data field.
+    #   // external_data stores key-value pairs describing data location. Recognized keys are:
+    #   // - "location" (required) - POSIX filesystem path relative to the directory where the ONNX
+    #   //                           protobuf model was stored
+    #   // - "offset" (optional) - position of byte at which stored data begins. Integer stored as string.
+    #   //                         Offset values SHOULD be multiples 4096 (page size) to enable mmap support.
+    #   // - "length" (optional) - number of bytes containing data. Integer stored as string.
+    #   // - "checksum" (optional) - SHA1 digest of file specified in under 'location' key.
+    #   repeated StringStringEntryProto external_data
+    location: str
+    offset: str     # !!! Consider using int, but int has int_max limitation
+    length: str     # !!! Consider using int, but int has int_max limitation
+    checksum: str
+
+
+@dataclass
+class Storage:
+    class DataLocation(Enum):
+        Internal = auto()
+        External = auto()
+
+    data_location: DataLocation
+    data: Union[Buffer, ExternalBuffer]
+
+
+# This is a concrete tensor backed by storage
+@dataclass
+class Tensor:
+    # storage
+    storage: Storage
+
+    # metadata
+    meta: TensorMeta
+
+
+################################################################################
+# Following section is the defining the schema of 3 level construct: GraphModule, Graph, Node
+
+# IValue has no corresponding class in fx
+# IValue is the "values" that are passed between nodes in the graph
+# !!! Assumption: Only Tensor can be passed between nodes, and not other int/float/bool types...
+# IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor
+# !!! Consider using a more descriptive name, e.g. TensorValue, TensorPlaceholder, TensorArgument, etc.
+@dataclass
+class IValue:
```

Comment:
@SherlockNoMad Can I treat IValue as identifier + FakeTensor like meta annotation for tensors?

Reply:
Per comment: 
IValue is a named virtual tensor, with an optional TensorMeta that describes the properties of the tensor